modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultConfigurationExpirationMonitor.java

```diff
@@ -0,0 +1,201 @@
+
+        private Builder()
+        {
+        }
+
+        /**
+         * Specifies how ofter should this instance check for expired config instances
+         *
+         * @param frequency a scalar time value
+         * @param timeUnit  a {@link TimeUnit} which qualifies the {@code frequency}
+         * @return {@code this} instance
+         */
+        public Builder runEvery(long frequency, TimeUnit timeUnit)
+        {
+            manager.frequency = frequency;
+            manager.timeUnit = timeUnit;
+
+            return this;
+        }
+
+        /**
+         * Provides a {@link BiConsumer} which receives each expired config instance and provides
+         * the behavior of how to expire it. The {@link BiConsumer} will receive the instance registration
+         * key as the first argument and the instance itself as the second
+         *
+         * @param expirationHandler a {@link BiConsumer} which acts as a expiration handler
+         * @return {@code this} instance
+         */
+        public Builder onExpired(BiConsumer<String, Object> expirationHandler)
+        {
+            manager.expirationHandler = expirationHandler;
+            return this;
+        }
+
+        /**
+         * Validates the provided configuration and returns an actual {@link ConfigurationExpirationMonitor}.
+         *
+         * @return a {@link ConfigurationExpirationMonitor}
+         */
+        public ConfigurationExpirationMonitor build()
+        {
+            checkArgument(manager.extensionRegistry != null, "extensionRegistry cannot be null");
+            checkArgument(manager.muleContext != null, "muleContext cannot be null");
+            checkArgument(manager.frequency > 0, "frequency must be greater than zero");
+            checkArgument(manager.timeUnit != null, "timeUnit cannot be null");
+
+            return manager;
+        }
+
+    }
+
+    private ExtensionRegistry extensionRegistry;
+    private MuleContext muleContext;
+    private long frequency;
+    private TimeUnit timeUnit;
+    private BiConsumer<String, Object> expirationHandler;
+
+    private ScheduledExecutorService executor;
+
+    private DefaultConfigurationExpirationMonitor()
+    {
+    }
+
+    /**
+     * Starts a scheduler which fires expiration tasks on the given {@link #frequency} and executes
+     * the {@link #expirationHandler} on each matching configuration instance
+     *
+     * @throws MuleException if the manager could not be started
+     */
+    @Override
+    public void start() throws MuleException
+    {
+        executor = Executors.newSingleThreadScheduledExecutor(r -> new Thread(r, getPrefix(muleContext) + "extension.expiration.manager"));
+        executor.scheduleWithFixedDelay(() -> expire(), frequency, frequency, timeUnit);
+    }
+
+    private void expire()
+    {
+        if (muleContext.isStopping() || muleContext.isStopped())
```

Comment:
The end result would be the same because this is done in a way that if the mule context starts stopping while this task is running, it will wait for it to finish.

However, checking on each iteration is a good optimization. I'll do that